### ISSUE TITLE
This `test/src/rules/extensions.js` test should be failing with 2 errors not 1

### ISSUE THREAD

**Issue by millerized** [benmosher/eslint-plugin-import#965]
This `invalid` test should fail with 2 errors – 1 for `.js` and another for `.jsx`.

https://github.com/benmosher/eslint-plugin-import/blob/a5844d57be0c3757248c7b824e4cc9c2c844db33/tests/src/rules/extensions.js#L129-L144

The error that I believe is missing:

```js
{
    message: 'Unexpected use of file extension "jsx" for "./bar.jsx"',
    line: 2,
    column: 22,
},

**Comment by ljharb** [comment]
We should probably audit all of the tests for this rule to see where the problem is.

**Comment by silvenon** [comment]
I believe this rule behaves correctly because `.js` is listed as the first extension under `settings > import/resolve > extensions`, meaning that `./bar` will always try to resolve first as `./bar.js`, which exists, so the `.jsx` should be necessary in order to target `bar.jsx` specifically.

The fact that you can't import twice from the same module should be outside the scope of this rule and resolving extension shouldn't depend on the import order.

**Comment by ljharb** [comment]
Hmm - if that's true, then a second test, that's an exact duplicate but without the first import line, would fail with just one error - the "missing" one referenced above.

**Comment by silvenon** [comment]
The error is not there, I'll try to fix it.

**Comment by silvenon** [comment]
Also without the first specified extension being `.js`.  This works:

```js
test({
  code: [
    'import component from "./bar.jsx"',
    'import data from "./bar.json"',
  ].join('\n'),
  options: [ { json: 'always', js: 'never', jsx: 'never' } ],
  settings: { 'import/resolve': { 'extensions': [ '.jsx', '.json', '.js' ] } },
  errors: [
    {
        message: 'Unexpected use of file extension "jsx" for "./bar.jsx"',
        line: 1,
        column: 23,
    },
  ],
}),
```

**Comment by silvenon** [comment]
This might seem like confusing behavior at first, but the alternative is impossible, so forcing the extension when omitting it would not yield the desired result is pretty smart.

**Comment by ljharb** [comment]
Sounds like we might want to add more test cases, but everything's correct as-is?
